[slow sign in] denormalize last message

rasom opened the pull request:
implements 
> - Store the latest message (or part of it which is shown in chat list) along with chat. In this case no need to make extra query in order to show chats list properly. **C**

from https://notes.status.im/VxmgW_rJQkqHsPoqGiMGxQ

status: ready <!-- Can be ready or wip -->

ghost commented:
<!--prchecklist--> 
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://docs.status.im/docs/build_status.html))?

rasom committed:
[slow sign in] Denorlmalize last message

The last messages of the chats are necessary to properly show the chat
list, which is shown right after signing in. Before this commit, the
last message was retrieved as one of 20 last messages fetched for each
chat.

Implementation:
- `:last-message-content` and `:last-message-type` fields were added to
  `chat` entity
- both fields are updated when messages are received/sent
- loading of the last 20 messages for each chat was removed as
  initialization step

cammellos commented on src/status_im/data_store/realm/schemas/account/migrations.cljs:
```diff
@@ -298,3 +298,22 @@
+   (.sorted "timestamp" true)
+   (aget 0)))
+
+;; TODO: implement migrations
```
Is this still valid?

rasom replied:
nope, it is not. Removed, thanks

mandrigin reviewed (approved):
lgtm

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/2/) :small_blue_diamond: ~13 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: android package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-155425-cd6627-pr.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android-e2e/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/2/) :small_blue_diamond: ~13 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: android e2e package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181212-155424-cd6627-e2e.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios-e2e/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/2/) :small_blue_diamond: ~18 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: ios e2e package](null)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/macos/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/2/) :small_blue_diamond: ~18 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: macos package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-155430-cd6627-pr.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/2/) :small_blue_diamond: ~21 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: ios package](https://i.diawi.com/fvBGSJ)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/2/) :small_blue_diamond: ~22 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-155429-cd6627-pr.AppImage)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/windows/PR-7055#2](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/2/) :small_blue_diamond: ~22 min :small_blue_diamond: cd6627fc :small_blue_diamond: [:package: windows package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181212-155432-cd6627-pr.exe)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/3/) :small_blue_diamond: ~13 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: android package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082542-9e6a31-pr.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios-e2e/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/3/) :small_blue_diamond: ~14 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: ios e2e package](null)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/macos/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/3/) :small_blue_diamond: ~14 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: macos package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082545-9e6a31-pr.dmg)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/windows/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/3/) :small_blue_diamond: ~15 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: windows package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082546-9e6a31-pr.exe)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/3/) :small_blue_diamond: ~17 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: ios package](https://i.diawi.com/gH9Qe3)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/3/) :small_blue_diamond: ~17 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-082544-9e6a31-pr.AppImage)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android-e2e/PR-7055#3](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/3/) :small_blue_diamond: ~18 min :small_blue_diamond: 9e6a31c6 :small_blue_diamond: [:package: android e2e package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181213-082541-9e6a31-e2e.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/4/) :small_blue_diamond: ~14 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: android package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094740-3b5907-pr.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/android-e2e/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/4/) :small_blue_diamond: ~15 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: android e2e package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181213-094740-3b5907-e2e.apk)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios-e2e/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/4/) :small_blue_diamond: ~15 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: ios e2e package](null)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/windows/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/4/) :small_blue_diamond: ~16 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: windows package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094745-3b5907-pr.exe)

status-im-auto commented:
#### :x: [status-react/prs/macos/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/4/) :small_orange_diamond: ~19 min :small_orange_diamond: 3b5907bf :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/4//consoleText)

status-im-auto commented:
#### :x: [status-react/prs/ios/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/4/) :small_orange_diamond: ~22 min :small_orange_diamond: 3b5907bf :small_orange_diamond: [:page_facing_up: build log](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/4//consoleText)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/linux/PR-7055#4](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/4/) :small_blue_diamond: ~29 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: linux package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181213-094743-3b5907-pr.AppImage)

status-im-auto commented:
#### :heavy_check_mark: [status-react/prs/ios/PR-7055#5](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/5/) :small_blue_diamond: ~14 min :small_blue_diamond: 3b5907bf :small_blue_diamond: [:package: ios package](https://i.diawi.com/R8DTKc)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/5/) | 7:45:54 PM | ~12 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-193308-21970f-pr.dmg) |
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/5/) | 7:46:32 PM | ~13 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-193304-21970f-pr.apk) |
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/5/) | 7:46:58 PM | ~13 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181214-193305-21970f-e2e.app.zip) |
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/5/) | 7:49:04 PM | ~15 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-193309-21970f-pr.exe) |
| :heavy_check_mark: | 21970fe0 | [#6](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/6/) | 7:49:41 PM | ~16 min | `ios` | [:package: package](https://i.diawi.com/fEeF7V) |
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/5/) | 7:51:16 PM | ~18 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181214-193303-21970f-e2e.apk) |
| :heavy_check_mark: | 21970fe0 | [#5](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/5/) | 7:52:07 PM | ~18 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181214-193307-21970f-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/6/) | 6:57:26 AM | ~12 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-064436-251d9d-pr.dmg) |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/6/) | 6:58:49 AM | ~14 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181216-064433-251d9d-e2e.app.zip) |
| :heavy_check_mark: | 251d9dd2 | [#7](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/7/) | 6:58:57 AM | ~14 min | `ios` | [:package: package](https://i.diawi.com/2DcbbK) |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/6/) | 6:59:09 AM | ~14 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-064433-251d9d-pr.apk) |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/6/) | 7:02:27 AM | ~17 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-064436-251d9d-pr.exe) |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/6/) | 7:03:04 AM | ~18 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181216-064432-251d9d-e2e.apk) |
| :heavy_check_mark: | 251d9dd2 | [#6](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/6/) | 7:05:00 AM | ~20 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-064435-251d9d-pr.AppImage) |
| | | | | | | |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7055/7/) | 8:56:46 AM | ~11 min | `macos` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-084450-c074d7-pr.dmg) |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/ios-e2e/job/PR-7055/7/) | 9:00:00 AM | ~15 min | `ios-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181216-084447-c074d7-e2e.app.zip) |
| :heavy_check_mark: | c074d73d | [#8](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-7055/8/) | 9:01:49 AM | ~17 min | `ios` | [:package: package](https://i.diawi.com/V4mLdM) |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/android/job/PR-7055/7/) | 9:03:28 AM | ~18 min | `android` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-084446-c074d7-pr.apk) |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/7/) | 9:04:10 AM | ~19 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181216-084445-c074d7-e2e.apk) |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7055/7/) | 9:04:12 AM | ~19 min | `linux` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-084449-c074d7-pr.AppImage) |
| :heavy_check_mark: | c074d73d | [#7](https://ci.status.im/job/status-react/job/prs/job/windows/job/PR-7055/7/) | 9:07:13 AM | ~22 min | `windows` | [:package: package](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-181216-084451-c074d7-pr.exe) |
| | | | | | | |
| :heavy_check_mark: | c440b7a3 | [#8](https://ci.status.im/job/status-react/job/prs/job/android-e2e/job/PR-7055/8/) | 11:48:39 AM | ~19 min | `android-e2e` | [:package: package](https://status-im.ams3.digitaloceanspaces.com/StatusIm-181217-112935-c440b7-e2e.apk) |
